gensim/models/hdpmodel.py

```diff
@@ -508,12 +511,16 @@ def update_finished(self, start_time, chunks_processed, docs_processed):
             If True - model is updated, False otherwise.
 
         """
+        try:
+            start_time = time.time()
+        except AttributeError:
+            start_time = time.clock()
         return (
             # chunk limit reached
             (self.max_chunks and chunks_processed == self.max_chunks)
 
             # time limit reached
-            or (self.max_time and time.clock() - start_time > self.max_time)
+            or (self.max_time and start_time - start_time > self.max_time)
```

Comment:
`start_time - start_time` makes no sense (and the above `AttributeError`-complicated fallback to `.clock()` seems fishy too, especially since the deprecation of `.clock()` is exactly the reason for this PR). But, I have a different fix incoming in #2715.